src/Persistence/Reflection/EnumReflectionProperty.php

```diff
@@ -32,31 +32,16 @@ public function __construct(ReflectionProperty $originalReflectionProperty, stri
         $this->enumType                   = $enumType;
     }
 
-    /**
-     * {@inheritDoc}
-     *
-     * @psalm-external-mutation-free
```

Comment:
PHPStan apparently doesn't understand that annotation, but then again, it does not complain when it is gone.

Reply:
I don't understand why it was used for those getters.

Reply:
phpstan memoizes getters by default unless tagged with `@phpstan-impure`, while psalm does not. This `@psalm-external-mutation-free` is required for Psalm to consider that calling a method does not mutate other things and so it can remember refinements it did for calls.